docs/_posts/2022-06-24-types-as-comments-proposal.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+title:  Types as comments for plain JavaScript
+date:   2022-06-16 08:00:13 +0100
+categories: typescript, javascript
+author: Friedrich Kurz
+---
+Today we're looking into the potential future of JavaScript by reading up on the types as comments proposal that seeks to introduce static type information as part of the core JavaScript language.
+
+## Types as comments
+
+[Types as comments][tc39-types-as-comments] was proposed by the [TC39][tc39] technical committee for JavaScript in late 2020. It proposes the addition of syntax for **type information** as a core, but optional, part of the JavaScript language. 
+
+Adding type information syntax to the language enables the potential use of static type checkers to prevent certain classes of bugs. That is to say, static type checking would be possible without having to develop in another statically typed language (such as *TypeScript* or *Flow*) that transpiles to JavaScript. 
+
+For JavaScript developers, this promises at least two things: 
+1. a faster development loop (i.e. without a transpilation step), and
+2. free choice of tooling for type checking.
+
+## Why static type checking?
+
+### Cannot read property 'bar' of undefined
+
+The case for static type checking is not particularly hard.
+
+As JavaScript developers, we probably all know the language's famously weak and dynamic type system which has no problem with letting funny code like the following pass to the interpreter without throwing errors (all examples were run with Node v14.19.3)
+
+```bash
+$ node -e 'console.log(3 + [])'
+3
+$ node -e 'console.log({}.foo)' 
+undefined
+```
+
+> :information_source: Have a look at [Programming TypeScript][oreilly-programming-typescript] (p. 3) for more examples.
```

Comment:
You might want to deep link to p3 to make it easier to find the correct examples:

https://www.google.de/books/edition/Programming_TypeScript/Y-mUDwAAQBAJ?hl=en&gbpv=1&pg=PA3&printsec=frontcover